### PR TITLE
[onert] Add unit tests for training Pad op

### PR DESCRIPTION
**Pull request by ragmani**
This commit adds some tests to validate training of Pad op.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
+
+    _context = std::make_unique<GenModelTrainContext>(cgen.finish());
+    _context->addTrainCase(
+      uniformTCD<float>({{{0, 1, 2, 3, 4, 5, 6, 7}}, {{7, 6, 5, 4, 3, 2, 1, 0}}}, // inputs
+                        {{{0, 13, 52, 0}}, {{0, 31, 24, 0}}},                     // expected
+                        {1.3900f}                                                 // loss
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc:47]:
```
Epoch 1/5
2/2 [==============================] - 0s 3ms/step - loss: 462.7862 - categorical_accuracy: 0.0000e+00
Epoch 2/5
2/2 [==============================] - 0s 1ms/step - loss: 32.8115 - categorical_accuracy: 0.5000
Epoch 3/5
2/2 [==============================] - 0s 1ms/step - loss: 5.8401 - categorical_accuracy: 1.0000
Epoch 4/5
2/2 [==============================] - 0s 1ms/step - loss: 1.3900 - categorical_accuracy: 1.0000
Epoch 5/5
2/2 [==============================] - 0s 1ms/step - loss: 0.3422 - categorical_accuracy: 1.0000
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    int output = cgen.addTensor({{1, 4}, circle::TensorType::TensorType_FLOAT32});
+    cgen.addOperatorFullyConnected({{input, weight, bias}, {fc_output}});
+    cgen.addOperatorPad({{fc_output, padding}, {output}});
+    cgen.setInputsAndOutputs({input}, {output});
+
+    float learning_rate = 0.01f;
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc:41]: If #13199 is merged first,
```suggestion
                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
    cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  // Invalid shape: output shape should be equal to input shape
+  int output = cgen.addTensor({{1, 2, 3, 1}, circle::TensorType::TensorType_FLOAT32});
+  cgen.addOperatorPad({{input, padding}, {output}});
+  cgen.setInputsAndOutputs({input, padding}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc:73]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+{
+  CirclePlusGen cgen;
+  std::vector<int32_t> padding_data{0, 0, 1, 1, 1, 1, 0, 0};
+  uint32_t padding_buf = cgen.addBuffer(padding_data);
+  // Invalid  type: input tensor type should be FLOAT32
+  int input = cgen.addTensor({{1, 2, 2, 1}, circle::TensorType::TensorType_INT32});
+  int padding = cgen.addTensor({{4, 2}, circle::TensorType::TensorType_INT32, padding_buf});
+  int output = cgen.addTensor({{1, 4, 4, 1}, circle::TensorType::TensorType_FLOAT32});
+  cgen.addOperatorPad({{input, padding}, {output}});
+  cgen.setInputsAndOutputs({input, padding}, {output});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Pad.test.cc:90]:
```suggestion
  cgen.setInputsAndOutputs({input, padding}, {output});
  cgen.markAllOpsAsTrainable();
```

**Comment by hseok-oh** [comment]
@ragmani https://github.com/Samsung/ONE/pull/13199 is merged

**Comment by ragmani** [comment]
> @ragmani https://github.com/Samsung/ONE/pull/13199 is merged

Added it :)

**Comment by jyoungyun** [comment]
@nnfw-bot test onert-x64-release

**Comment by ragmani** [comment]
@Samsung/one_onert  PTAL

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]

**Pull request merged**